devito/types/sparse.py

```diff
@@ -1048,17 +1048,20 @@ def __init__(self, *args, **kwargs):
             gridpoints.data[:] = gridpoints_data[:]
             self._gridpoints = gridpoints
 
-            coefficients = SubFunction(name="%s_coefficients" % self.name,
-                                       dimensions=(self.indices[-1], Dimension(name='d'),
-                                                   Dimension(name='i')),
-                                       shape=(self.npoint, self.grid.dim, self.r),
-                                       dtype=self.dtype, space_order=0, parent=self)
-            coefficients_data = kwargs.get('coefficients', None)
+            interpolation_coeffs = SubFunction(name="%s_interpolation_coeffs" % self.name,
```

Comment:
does this get less ugly with ``interp_coeffs``?

Reply:
For developers yes, but pre computed sparse functions still requires a public `interpolation_coeffs` interface - If you think `interp_coeffs` is fine for a public name I'm happy to change.